ui/pages/confirmations/confirmation/confirmation.js

```diff
@@ -290,6 +294,10 @@ export default function ConfirmationPage({
           {
             matchedChain,
             currencySymbolWarning,
+            existingNetworkConfiguration:
+              networkConfigurationsByChainId?.[
+                pendingConfirmation.requestData?.chainId
+              ],
```

Comment:
`ui/pages/confirmations/confirmation/templates/add-ethereum-chain.js` needs to know whether a network already exists for the chain, to customize the UI message.  It doesn't feel right to push it down like this, but I couldn't find another way.